### PR TITLE
Add support for key-value lists (kvlists)

### PR DESCRIPTION
**Pull request by andreashasse**
Implements recognition and JSON encoding/decoding of lists of 2-tuples as key-value lists, which serialize to/from JSON objects.

Features:
- Recognizes [{KeyType, ValueType}] patterns as kvlists in type specs
- Supports multiple key types: atoms, binaries, strings, integers, floats
- Supports any value type that spectra handles (maps, lists, nested kvlists, etc.)
- Proper error handling for unsupported key types (maps, lists)
- Type validation checks key TYPE not key VALUE to catch errors early

Implementation:
- Added sp_kvlist record to internal types
- Enhanced spectra_abstract_code to detect 2-tuple list patterns
- Added kvlist_to_json/4 and kvlist_from_json/4 in spectra_json
- Added is_json_compatible_key_type/1 validator

Tests (24 new tests, all passing):
- Comprehensive coverage of all supported key types
- Complex value types (maps, lists, nested kvlists)
- Error handling for invalid key types
- Edge cases (empty lists, type mismatches)

All 356 tests passing.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/spectra_json.erl [hunk 1]
```diff
@@ -187,6 +191,85 @@ list_to_json(TypeInfo, Type, Data) when is_list(Data) ->
...
+-spec kvlist_to_json(
+    TypeInfo :: spectra:type_info(),
+    KeyType :: spectra:sp_type_or_ref(),
+    ValType :: spectra:sp_type_or_ref(),
+    Data :: [{term(), term()}]
+) ->
+    {ok, json:encode_value()} | {error, [spectra:error()]}.
+kvlist_to_json(TypeInfo, KeyType, ValType, Data) when is_list(Data) ->
+    %% First, validate that the key type is JSON-compatible
+    case is_json_compatible_key_type(KeyType) of
```
Comment (andreashasse) [inline src/spectra_json.erl]: Not sure if we should validate this or just crash?

File: test/kvlist_test.erl [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+-include_lib("eunit/include/eunit.hrl").
+
+-include("../include/spectra.hrl").
+
+%% Key-value list types with different key types
+-type string_kvlist() :: [{string(), integer()}].
+-type atom_kvlist() :: [{atom(), binary()}].
+-type binary_kvlist() :: [{binary(), string()}].
+-type integer_kvlist() :: [{integer(), atom()}].
+-type float_kvlist() :: [{float(), binary()}].
```
Comment (andreashasse) [inline test/kvlist_test.erl]: Is this a valid kvlist? `[{atom(), atom()}|{integer(), integer()}]`?
